File: frontend/web_ccadmin_lt/src/app/enterprise/shared/model/entity/StoreEntity.ts
import { AuditTableEntity } from "./AuditTableEntity";

export class StoreEntity extends AuditTableEntity {
    
    public StoreCod: string;
    public Name: string;
    public Description: string;
    public Address: string;
    public UbigeoCod: string;
    public SunatAddressTypeCode: string;
    public IsVirtualStoreEnabled: string;
    public Latitude: number | null;
    public Longitude: number | null;
    public CompanyCod: string;
    public CountryCod: string;

    constructor() {
        super();
        this.StoreCod = '';
        this.Name = '';
        this.Description = '';
        this.Address = '';
        this.UbigeoCod = '';
        this.SunatAddressTypeCode = '0000';
        this.IsVirtualStoreEnabled = 'N';
        this.Latitude = null;
        this.Longitude = null;
        this.CompanyCod = '';
        this.CountryCod = 'PER';
    }
}

File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/store/model/entity/StoreEntity.java
package com.ccadmin.app.store.model.entity;

import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.io.Serializable;
import java.math.BigDecimal;

@Entity
@Table(name = "store")
public class StoreEntity extends AuditTableEntity implements Serializable {

    @Id
    public String StoreCod;
    public String Name;
    public String Description;
    public String Address;
    public String UbigeoCod;
    public String SunatAddressTypeCode = "0000";
    public String IsVirtualStoreEnabled = "N";
    public BigDecimal Latitude;
    public BigDecimal Longitude;
    public String CompanyCod;
    public String CountryCod = "PER";
}
